Lib/test/test_traceback.py

```diff
@@ -92,6 +92,58 @@ def test_base_exception(self):
         lst = traceback.format_exception_only(e.__class__, e)
         self.assertEqual(lst, ['KeyboardInterrupt\n'])
 
+    def test_traceback_context_recursionerror(self):
+        # Test that for long traceback chains traceback does not itself
+        # raise a recursion error while printing (Issue43048)
+
+        # Calling f() creates a stack-overflowing __context__ chain.
+        def f():
+            try:
+                raise ValueError('hello')
+            except ValueError:
+                f()
+
+        try:
+            f()
+        except RecursionError:
+            exc_info = sys.exc_info()
+            traceback.format_exception(exc_info[0], exc_info[1], exc_info[2])
+
+    def test_traceback_cause_recursionerror(self):
+        # Same as test_traceback_context_recursionerror, but with
+        # a __cause__ chain.
+
+        def f():
+            e = None
+            try:
+                f()
+            except Exception as exc:
+                e = exc
+            raise Exception from e
+
+        try:
+            f()
+        except Exception:
+            exc_info = sys.exc_info()
+            traceback.format_exception(exc_info[0], exc_info[1], exc_info[2])
```

Comment:
For the purpose of testing the traceback methods it would be ok to unindent the `traceback.format_exception(exc_info[0], exc_info[1], exc_info[2])` line so that it's not happening in the except block.

Reply:
done